IPython/testing/iptest.py

```diff
@@ -58,6 +59,24 @@
 warnings.filterwarnings('ignore', 'wxPython/wxWidgets release number mismatch',
                         UserWarning)
 
+# Enable printing all warnings raise by IPython's modules
+warnings.filterwarnings('default', message='.*', category=Warning, module='IPy.*')
+
+
+if version_info < (6,):
```

Comment:
Will 4.2 force user to upgrade Traitlets ? I was thinking that for all 4.x required version would not changed, 
that 5.0 would mostly migrate to new API, keeping (some) backward shims + deprecation warnings, and 6.0 actually use new versions. I'm fine changing for earlier versions.